### PR TITLE
User cancels an order userstory30 #45

### LINKED ISSUE #45: User Story 30, User cancels an order
```
As a registered user
When I visit an order's show page
If the order is still "pending", I see a button or link to cancel the order
When I click the cancel button for an order, the following happens:
- Each row in the "order items" table is given a status of "unfulfilled"
- The order itself is given a status of "cancelled"
- Any item quantities in the order that were previously fulfilled have their quantities returned to their respective merchant's inventory for that item.
- I am returned to my profile page
- I see a flash message telling me the order is now cancelled
- And I see that this order now has an updated status of "cancelled"
```

DO THESE STORIES FIRST:
- User Story 17, User Profile Show Page
- User Story 29, User views an Order Show Page

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #49

child of #82

depends on #60

depends on #46

### PR DESCRIPTION
**Pull request by milevy1**
- [] Check this if the PR has been approved by the team to be a quick patch and the rest of this form will not be filled out

# Test Coverage Percentage (Model/Feature):
- rspec - 100%
- rspec spec/models - x%

# What functionality does this accomplish?
User Story 30, User cancels an order
- I will add to the issues list that this Order#update controller action should be refactored to model methods.  In the interest of time, I left it as is. I left lots of comments in the code explaining each step though.

# What did you struggle on to complete?

# Implements/Fixes:
* description
closes #45 

# Testing Changes
- [x] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

# Please include an emoji of how you feel about this branch:
#  😽

### CODE REVIEW

File: spec/features/reg_users/cancel_order_spec.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      @merchant = create(:merchant)
+      @item_1 = create(:item, user: @merchant, quantity: 10)
+      @item_2 = create(:item, user: @merchant, quantity: 10)
+
+      @user = create(:user)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user)
+      @order = create(:order, user: @user, status: 0)
+
+      @unfulfilled_order_item = create(:order_item, order: @order, item: @item_1, quantity: 1)
+      @fulffileed_order_item = create(:fulfilled_order_item, order: @order, item: @item_2, quantity: 1)
```
Comment (blake-enyart) [inline spec/features/reg_users/cancel_order_spec.rb]: *fulfilled_order_item

File: spec/features/reg_users/cancel_order_spec.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+      visit order_path(@order)
+    end
+
+    scenario 'if the order is pending, I see a link to cancel the order' do
+      expect(page).to have_link("Cancel Order")
+    end
+
+    scenario 'if the order is NOT pending, I do not see a link to cancel the order' do
+      @packaged_order = create(:packaged_order)
```
Comment (blake-enyart) [inline spec/features/reg_users/cancel_order_spec.rb]: I would recommend not using instance variable instantiation in individual test in case there is overlap in future test in this same describe block given the length of these test files.

File: spec/features/reg_users/cancel_order_spec.rb (hunk 1 above)
Comment (milevy1) [reply spec/features/reg_users/cancel_order_spec.rb]: Nice catch, corrected.

File: spec/features/reg_users/cancel_order_spec.rb (hunk 2 above)
Comment (milevy1) [reply spec/features/reg_users/cancel_order_spec.rb]: Agreed. Changed them to local variables.
